docs/api-documentation-guide.md

```diff
@@ -1,126 +1,142 @@
--  Capitalize standard words like URL, HTTP, or JSON.
-   -  Incorrect: # Creates http server endpoints.
-   -  Correct:   # Creates HTTP server endpoints.
--  Add the Oxford comma before “and” or “or” in lists.
-   -  Incorrect: # + message - An HTTP outbound request message or any payload of type `string`, `xml`, `json`, `byte[]`, `io:ReadableByteChannel` or `mime:Entity[]`
-   -  Correct: # + message - An HTTP outbound request message or any payload of type `string`, `xml`, `json`, `byte[]`, `io:ReadableByteChannel`, or `mime:Entity[]`
-   -  Note: We do not need the Oxford comma when there are only 2 items in the list as follows. E.g.,
-
-```
-foo or bar
-foo, bar, or baz
-foo and bar
-foo, bar, and baz
-```
--  Capitalize all file extensions.
-   -  Examples: BAL, ZIP
--  For names of third party technologies, use the exact way they write it (Google and check). 
-   -  Examples: MySQL not Mysql 
--  Add one line code snippet if possible. E.g., https://ballerina.io/v1-2/learn/api-docs/ballerina/java/index.html#annotations.
--  All the referrings of records/objects/functions inside the API docs and Module.md should have the representation as `<module-name>:<type-name>` instead of `<type-name>`.
--  Any description related to the parameter goes beyond a single line, and should start with the starting position of the 1st line.
-   - Example: # + httpClient - Chain of different HTTP clients which provides the capability   
-    \# 		   for initiating contact with a remote HTTP service in resilient  
-    \# 		   manner
--  Do not use the word “users” in the comments. Instead, address the users directly using the word “you”. E.g.,
-   -  Wrong: Users can also pass a key/value pair where the value is an error stack trace.
-   -  Right: You can also pass a key/value pair where the value is an error stack trace.
+5. You should NOT try to refer to the BBEs with links from within API Docs, but rather have separate examples. This is because the required examples will probably not match the examples in the BBEs. The API Docs should have more examples to explain each API operation etc. Also, you will not want to click another link and navigate away from the API docs to a separate BBE page. Rather, you will need to see the example in the same place where the module is described. 
 
-## Anti-Patterns
-[This doc](https://docs.google.com/document/d/1y6QVqaZzZt9jMpYV4jP5WRS_W_KoC4y40Uuoh1ALu8E/edit?usp=sharing) basically explains how to write better Ballerina code.
+6. The examples in API Docs should be self-contained. They should contain a fully working program when an example is given, for example, including all the imports and a main function. The Ballerina documentation generator can later remove any redundant code segments in a post-processing operation. This is required since this code should be executable from the Ballerina website. 
 
```

Comment:
I am not sure if this is a requirement. API docs usually have code snippets as examples. We cannot have fully executable code snippets in it, IMO. Because it will make the code two verbose. Why is it needed to be executed from the website ?

Reply:
+1 Removed that point.